### ISSUE TITLE
New lint: suggest `is_ok()` when using wild pattern in both Ok and Err

### ISSUE THREAD

**Issue by CYBAI** [rust-lang-nursery/rust-clippy#3290]
As @killercup mentioned in https://github.com/rust-lang-nursery/rust-clippy/pull/3281#discussion_r223504940 , we can have a new lint to suggest developers to change like following

```rust
// As is
match Ok::<i32, i32>(42) {
  Ok(_) => $1,
  Err(_) => $2,
};

// To be
if Ok::<i32, i32>(42).is_ok() { $1 } else { $2 }
```

Besides, I think this lint should also check the case of `Some` and `None`

```rust
// As is
match Some(42) {
  Some(_) => $1,
  None => $2
};

// To be
if Some(42).is_some() { $1 } else { $2 }
```

**Comment by killercup** [comment]
I think for the Option case at least I'd still prefer `if let Some(_name)` because changing this to a block that actually uses the value is only a very small change.

**Comment by mikerite** [comment]
There is already lint for this [single_match_else](https://rust-lang-nursery.github.io/rust-clippy/master/index.html#single_match_else).

**Comment by CYBAI** [comment]
@mikerite Got it! Let me update description to remove the `Option` part, thanks!

**Comment by nk9** [comment]
Seems like the request is to extend `single_match_else` to apply to `Result`, not just `Option`.

@rustbot label C-enhancement
